### PR TITLE
ddl: grammar check for create unsupported temporary table

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

We parsed almost all the temporary table related syntax in https://github.com/pingcap/parser/pull/1211
Some kinds of temporary table is not supported yet, so we need to handle the unsupported ones

Problem Summary:

### What is changed and how it works?

Enhance the `create temporary table` validation.

What's Changed:

- do not allow creating local temporary table (proper error message or warning)
- engine field should be empty or 'memory'

How it Works:

Add check and tests.

### Check List 

Tests 

- Unit test

### Release note 

- No release note

### CODE REVIEW

**Review by wjhuang2016 — APPROVED** [review]

File: ddl/ddl_api.go [hunk 1]
```diff
@@ -2217,6 +2223,12 @@ func handleTableOptions(options []*ast.TableOption, tbInfo *model.TableInfo) err
 			tbInfo.PreSplitRegions = op.UintValue
 		case ast.TableOptionCharset, ast.TableOptionCollate:
 			// We don't handle charset and collate here since they're handled in `getCharsetAndCollateInTableOption`.
+		case ast.TableOptionEngine:
+			if tbInfo.TempTableType != model.TempTableNone {
+				if op.StrValue != "" && op.StrValue != "memory" {
```
Comment (djshow832) [inline ddl/ddl_api.go:2228]: Is `op.StrValue` always in lower case?

File: ddl/error.go [hunk 2]
```diff
@@ -277,4 +277,7 @@ var (
 
 	// ErrPartitionNoTemporary returns when partition at temporary mode
 	ErrPartitionNoTemporary = dbterror.ClassDDL.NewStd(mysql.ErrPartitionNoTemporary)
+
+	errUnsupportedOnCommitPreserve = dbterror.ClassDDL.NewStdErr(mysql.ErrUnsupportedDDLOperation, parser_mysql.Message("TiDB doesn't support ON COMMIT PRESERVE ROWS for now", nil))
+	errUnsupportedEngineTemporary  = dbterror.ClassDDL.NewStdErr(mysql.ErrUnsupportedDDLOperation, parser_mysql.Message("TiDB doesn't this kind of engine for temporary table", nil))
```
Comment (djshow832) [inline ddl/error.go:282]:
```suggestion
	errUnsupportedEngineTemporary  = dbterror.ClassDDL.NewStdErr(mysql.ErrUnsupportedDDLOperation, parser_mysql.Message("TiDB doesn't support this kind of engine for temporary table", nil))
```

File: ddl/db_integration_test.go [hunk 3]
```diff
@@ -2803,3 +2803,25 @@ func (s *testIntegrationSuite3) TestIssue21835(c *C) {
...
 }
+
+func (s *testIntegrationSuite3) TestCreateTemporaryTable(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t;")
+
+	// Grammar error.
+	tk.MustGetErrCode("create global temporary table t(a double(0, 0))", errno.ErrParse)
+	tk.MustGetErrCode("create temporary table t(id int) on commit delete rows", errno.ErrParse)
```
Comment (djshow832) [inline ddl/db_integration_test.go:2814]: How about creating normal tables with `ON COMMIT` clause?

Comment (tiancaiamao) [reply ddl/db_integration_test.go:2814]: Done

File: ddl/error.go (hunk 2 above)
Comment (tiancaiamao) [reply ddl/error.go]: Done

File: ddl/ddl_api.go (hunk 1 above)
Comment (tiancaiamao) [reply ddl/ddl_api.go]: Fixed

**Pull request merged**